@darranl WFP test fails with:

```
2022-04-21 20:41:12,453 ERROR [org.jboss.as.controller.management-operation] (ServerService Thread Pool -- 67) WFLYCTL0013: Operation ("add") failed - address: ([("subsystem" => "microprofile-jwt-smallrye")]): java.lang.ExceptionInInitializerError
	at org.wildfly.extension.microprofile.jwt-smallrye@27.0.0.Beta1-SNAPSHOT//org.wildfly.extension.microprofile.jwt.smallrye.MicroProfileJWTSubsystemAdd.performBoottime(MicroProfileJWTSubsystemAdd.java:42)
	at org.jboss.as.controller@19.0.0.Beta7//org.jboss.as.controller.AbstractBoottimeAddStepHandler.performBoottime(AbstractBoottimeAddStepHandler.java:144)
	at org.jboss.as.controller@19.0.0.Beta7//org.jboss.as.controller.AbstractBoottimeAddStepHandler.performRuntime(AbstractBoottimeAddStepHandler.java:119)
	at org.jboss.as.controller@19.0.0.Beta7//org.jboss.as.controller.AbstractAddStepHandler$1.execute(AbstractAddStepHandler.java:164)
	at org.jboss.as.controller@19.0.0.Beta7//org.jboss.as.controller.AbstractOperationContext.executeStep(AbstractOperationContext.java:1045)
	at org.jboss.as.controller@19.0.0.Beta7//org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:777)
	at org.jboss.as.controller@19.0.0.Beta7//org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:466)
	at org.jboss.as.controller@19.0.0.Beta7//org.jboss.as.controller.ParallelBootOperationStepHandler$ParallelBootTask.run(ParallelBootOperationStepHandler.java:384)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1348)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.JBossThread.run(JBossThread.java:513)
Caused by: java.lang.IllegalArgumentException: Invalid logger interface org.wildfly.extension.microprofile.jwt.smallrye._private.MicroProfileJWTLogger (implementation not found in ModuleClassLoader for Module "org.wildfly.extension.microprofile.jwt-smallrye" version 27.0.0.Beta1-SNAPSHOT from local module loader @73302995 (finder: local module finder @1838ccb8 (roots: /opt/buildAgent/work/dc9a11b6a14b0d86/ee-9/dist/target/wildfly-preview-27.0.0.Beta1-SNAPSHOT/modules,/opt/buildAgent/work/dc9a11b6a14b0d86/ee-9/dist/target/wildfly-preview-27.0.0.Beta1-SNAPSHOT/modules/system/layers/base,/opt/buildAgent/work/dc9a11b6a14b0d86/testsuite/integration/web/target/modules)))
	at org.jboss.logging@3.4.3.Final//org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2564)
	at org.jboss.logging@3.4.3.Final//org.jboss.logging.Logger.getMessageLogger(Logger.java:2530)
	at org.jboss.logging@3.4.3.Final//org.jboss.logging.Logger.getMessageLogger(Logger.java:2516)
	at org.wildfly.extension.microprofile.jwt-smallrye@27.0.0.Beta1-SNAPSHOT//org.wildfly.extension.microprofile.jwt.smallrye._private.MicroProfileJWTLogger.<clinit>(MicroProfileJWTLogger.java:40)
	... 14 more
```

--- a/ee-9/common-microprofile/pom.xml
+++ b/ee-9/common-microprofile/pom.xml
@@ -52,6 +52,10 @@
         </dependency>
         <dependency>
             <groupId>${full.maven.groupId}</groupId>
+            <artifactId>wildfly-microprofile-jwt-smallrye-jakarta</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${full.maven.groupId}</groupId>
             <artifactId>wildfly-microprofile-openapi-smallrye-jakarta</artifactId>
         </dependency>
         <dependency>

--- a/ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml
+++ b/ee-9/common-microprofile/src/main/resources/license/preview-feature-pack-common-microprofile-licenses.xml
@@ -25,6 +25,17 @@
         </dependency>
         <dependency>
             <groupId>${full.maven.groupId}</groupId>
+            <artifactId>wildfly-microprofile-jwt-smallrye-jakarta</artifactId>
+            <licenses>
+                <license>
+                    <name>GNU Lesser General Public License v2.1 or later</name>
+                    <url>http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html</url>
+                    <distribution>repo</distribution>
+                </license>
+            </licenses>
+        </dependency>
+        <dependency>
+            <groupId>${full.maven.groupId}</groupId>
             <artifactId>wildfly-microprofile-openapi-smallrye-jakarta</artifactId>
             <licenses>
                 <license>

--- a/ee-9/feature-pack/pom.xml
+++ b/ee-9/feature-pack/pom.xml
@@ -544,6 +544,8 @@
                                 <exclude>org.wildfly:wildfly-clustering-infinispan-extension\z</exclude>
                                 <exclude>org.wildfly:wildfly-clustering-web-infinispan\z</exclude>
 
+                                <exclude>org.wildfly.security:wildfly-elytron-jwt\z</exclude>
+
                                 <exclude>org.hornetq:.+\z</exclude>
                                 <exclude>org.apache.activemq:artemis-jakarta-client\z</exclude>
                                 <exclude>org.apache.activemq:artemis-jakarta-ra\z</exclude>

--- a/ee-9/feature-pack/pom.xml
+++ b/ee-9/feature-pack/pom.xml
@@ -544,6 +544,7 @@
                                 <exclude>org.wildfly:wildfly-clustering-infinispan-extension\z</exclude>
                                 <exclude>org.wildfly:wildfly-clustering-web-infinispan\z</exclude>
 
+                                <exclude>org.wildfly:wildfly-microprofile-jwt-smallrye\z</exclude>
                                 <exclude>org.wildfly.security:wildfly-elytron-jwt\z</exclude>
 
                                 <exclude>org.hornetq:.+\z</exclude>

--- a/ee-9/pom.xml
+++ b/ee-9/pom.xml
@@ -1542,7 +1542,17 @@
                     </exclusion>
                 </exclusions>
             </dependency>
-
+            <dependency>
+                <groupId>${full.maven.groupId}</groupId>
+                <artifactId>wildfly-microprofile-jwt-smallrye-jakarta</artifactId>
+                <version>${project.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
             <dependency>
                 <groupId>${full.maven.groupId}</groupId>
                 <artifactId>wildfly-microprofile-openapi-smallrye-jakarta</artifactId>

--- a/ee-9/pom.xml
+++ b/ee-9/pom.xml
@@ -95,6 +95,7 @@
         <version.io.smallrye.smallrye-config>3.0.0-RC1</version.io.smallrye.smallrye-config>
         <version.io.smallrye.smallrye-fault-tolerance>6.0.0-RC2</version.io.smallrye.smallrye-fault-tolerance>
         <version.io.smallrye.smallrye-health>4.0.0-RC1</version.io.smallrye.smallrye-health>
+        <version.io.smallrye.smallrye-jwt>4.0.0-RC1</version.io.smallrye.smallrye-jwt>
         <version.io.smallrye.smallrye-metrics>4.0.0-RC1</version.io.smallrye.smallrye-metrics>
         <version.io.smallrye.smallrye-opentracing>3.0.0-RC1</version.io.smallrye.smallrye-opentracing>
         <!-- MP5 - END -->

--- a/ee-9/pom.xml
+++ b/ee-9/pom.xml
@@ -118,6 +118,7 @@
         <version.org.jboss.weld.weld-api>5.0.Beta5</version.org.jboss.weld.weld-api>
         <version.org.jvnet.staxex>2.0.1</version.org.jvnet.staxex>
         <version.org.wildfly.security.elytron>2.0.0.Beta1</version.org.wildfly.security.elytron>
+        <version.org.wildfly.security.elytron-mp>2.0.0.Beta2</version.org.wildfly.security.elytron-mp>
         <version.org.wildfly.security.elytron-web>2.0.0.Beta2</version.org.wildfly.security.elytron-web>
         <version.org.wildfly.security.jakarta.elytron-ee>2.0.0.Beta2</version.org.wildfly.security.jakarta.elytron-ee>
 

--- a/ee-9/source-transform/microprofile/jwt-smallrye/pom.xml
+++ b/ee-9/source-transform/microprofile/jwt-smallrye/pom.xml
@@ -1,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2022, Red Hat, Inc., and individual contributors
+  ~ as indicated by the @author tags. See the copyright.txt file in the
+  ~ distribution for a full listing of individual contributors.
+  ~
+  ~ This is free software; you can redistribute it and/or modify it
+  ~ under the terms of the GNU Lesser General Public License as
+  ~ published by the Free Software Foundation; either version 2.1 of
+  ~ the License, or (at your option) any later version.
+  ~
+  ~ This software is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  ~ Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public
+  ~ License along with this software; if not, write to the Free
+  ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.wildfly</groupId>
+        <artifactId>wildfly-microprofile-jakarta</artifactId>
+        <!--
+        Maintain separation between the artifact id and the version to help prevent
+        merge conflicts between commits changing the GA and those changing the V.
+        -->
+        <version>27.0.0.Beta1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>wildfly-microprofile-jwt-smallrye-jakarta</artifactId>
+
+    <name>WildFly: MicroProfile JWT Extension With SmallRye (Jakarta Namespace)</name>
+
+    <properties>
+        <transformer-input-dir>${project.basedir}/../../../../microprofile/jwt-smallrye</transformer-input-dir>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.jboss</groupId>
+            <artifactId>jboss-dmr</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss</groupId>
+            <artifactId>jandex</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss</groupId>
+            <artifactId>staxmapper</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.metadata</groupId>
+            <artifactId>jboss-metadata-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.metadata</groupId>
+            <artifactId>jboss-metadata-common</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.modules</groupId>
+            <artifactId>jboss-modules</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.msc</groupId>
+            <artifactId>jboss-msc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.core</groupId>
+            <artifactId>wildfly-controller</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.core</groupId>
+            <artifactId>wildfly-server</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>${ee.maven.groupId}</groupId>
+            <artifactId>wildfly-web-common</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging-annotations</artifactId>
+            <!-- This is a compile-time dependency of this project, but is not needed at compile or runtime by other
+      projects that depend on this project.-->
+            <scope>provided</scope>
+            <optional>true</optional>
+        </dependency>
+
+        <!-- Test Dependencies -->
+        <dependency>
+            <groupId>org.jboss.xnio</groupId>
+            <artifactId>xnio-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.projectodd.vdx</groupId>
+            <artifactId>vdx-wildfly</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.core</groupId>
+            <artifactId>wildfly-subsystem-test</artifactId>
+            <type>pom</type>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+</project>

--- a/ee-9/source-transform/microprofile/pom.xml
+++ b/ee-9/source-transform/microprofile/pom.xml
@@ -44,6 +44,7 @@
     <modules>
         <module>fault-tolerance-smallrye</module>
         <module>health-smallrye</module>
+        <module>jwt-smallrye</module>
         <module>openapi-smallrye</module>
         <module>reactive-streams-operators-smallrye/cdi-provider</module>
         <module>opentracing-extension</module>

--- a/microprofile/galleon-common/src/main/resources/modules/system/layers/base/org/wildfly/extension/microprofile/jwt-smallrye/main/module.xml
+++ b/microprofile/galleon-common/src/main/resources/modules/system/layers/base/org/wildfly/extension/microprofile/jwt-smallrye/main/module.xml
@@ -25,7 +25,7 @@
     </properties>
 
     <resources>
-        <artifact name="${org.wildfly:wildfly-microprofile-jwt-smallrye}"/>
+        <artifact name="\${org.wildfly:wildfly-microprofile-jwt-smallrye@module.jakarta.suffix@}"/>
     </resources>
 
     <dependencies>

--- a/pom.xml
+++ b/pom.xml
@@ -512,6 +512,7 @@
         <version.org.wildfly.extras.creaper>1.6.2</version.org.wildfly.extras.creaper>
         <version.org.wildfly.http-client>1.1.11.Final</version.org.wildfly.http-client>
         <version.org.wildfly.naming-client>1.0.15.Final</version.org.wildfly.naming-client>
+        <version.org.wildfly.security.elytron-mp>1.19.0.Final</version.org.wildfly.security.elytron-mp>
         <version.org.wildfly.transaction.client>2.0.1.Final</version.org.wildfly.transaction.client>
         <version.rhino.js>1.7R2</version.rhino.js>
         <version.sun.jaxb>2.3.3-b02</version.sun.jaxb>
@@ -9726,6 +9727,12 @@
                         <artifactId>*</artifactId>
                     </exclusion>
                 </exclusions>
+            </dependency>
+
+            <dependency>
+                <groupId>org.wildfly.security</groupId>
+                <artifactId>wildfly-elytron-jwt</artifactId>
+                <version>${version.org.wildfly.security.elytron-mp}</version>
             </dependency>
 
             <dependency>

